Epub/KoeBook.Epub/Services/ScrapingNaroService.cs

```diff
@@ -36,143 +48,67 @@ public async ValueTask<EpubDocument> ScrapingAsync(string url, string coverFileP
             {
-                List<SectionWithChapterTitle> SectionWithChapterTitleList = new List<SectionWithChapterTitle>();
-                for (int i = 1; i <= allNum; i++)
-                {
-                    await Task.Delay(1500, ct);
-                    var pageUrl = Path.Combine(url, i.ToString());
-                    var load = await ReadPageAsync(pageUrl, isRensai, imageDirectory, ct).ConfigureAwait(false);
-                    SectionWithChapterTitleList.Add(load);
-                }
-                string? chapterTitle = null;
-                foreach (var sectionWithChapterTitle in SectionWithChapterTitleList)
+                async IAsyncEnumerable<(string? title, Section section)> LoadDetailsAsync(IBrowsingContext context, NovelInfo novelInfo, string imageDirectory, [EnumeratorCancellation] CancellationToken ct)
                 {
-                    if (sectionWithChapterTitle == null)
-                        throw new EbookException(ExceptionType.WebScrapingFailed, "ページの取得に失敗しました");
-
-                    if (sectionWithChapterTitle.title != null)
+                    for (int i = 1; i <= novelInfo.GeneralAllNo; i++)
                     {
-                        if (sectionWithChapterTitle.title != chapterTitle)
+                        IDocument doc;
                         {
-                            chapterTitle = sectionWithChapterTitle.title;
-                            document.Chapters.Add(new Chapter() { Title = chapterTitle });
-                            document.Chapters[^1].Sections.Add(sectionWithChapterTitle.section);
-                        }
-                        else
-                        {
-                            document.Chapters[^1].Sections.Add(sectionWithChapterTitle.section);
+                            // htmlはローカルでキャプチャする
+                            var html = await _scrapingClientService.GetAsStringAsync($"{BaseUrl}/{ncode}/{i}", ct).ConfigureAwait(false);
+                            doc = await context.OpenAsync(request => request.Content(html), ct).ConfigureAwait(false);
                         }
+                        yield return await ReadPageAsync(doc, true, imageDirectory, ct).ConfigureAwait(false);
                     }
-                    else
+                }
+
+                string? chapterTitle = null;
+                await foreach (var (title, section) in LoadDetailsAsync(context, novelInfo, imageDirectory, ct).ConfigureAwait(false).WithCancellation(ct))
+                {
+                    if (title != null)
                     {
-                        if (document.Chapters.Count == 0)
-                        {
-                            document.Chapters.Add(new Chapter());
-                        }
-                        document.Chapters[^1].Sections.Add(sectionWithChapterTitle.section);
+                        if (title != chapterTitle)
+                            document.Chapters.Add(new Chapter() { Title = title });
```

Comment:
```suggestion
                        {
                            document.Chapters.Add(new Chapter() { Title = title });
                            chapterTitle = title;
                        }
```
Chapterを追加する前か後に、chapterTitleを更新する必要があります。